### DISCUSSION TITLE
Info on AWS metal instance

### DISCUSSION THREAD

**Discussion by rootfs** [sustainable-computing-io/kepler#1116] [Show and tell]
There have been some discussions on whether RAPL readings can be found on AWS metal instances. I took a step to check today.

Short story: on an `i3.metal` instance, I can read RAPL pkg and dram domains. But the default ubuntu 20.04 build doesn't come with RAPL kernel module, so I have to install `msr-tools` and run `rdmsr` command to get the readings. Based on this findings, we should keep the MSR pkg in kepler and allow [EnableMSR](https://github.com/sustainable-computing-io/kepler/blob/main/pkg/config/config.go#L65) to be turned on.
 
![image](https://github.com/sustainable-computing-io/kepler/assets/7062400/46de9659-f6db-4ab7-90d4-2066448beb05)

@jiere can you take a look? Thanks

This is also part of the process to turn on power model training on AWS metal instances
https://github.com/sustainable-computing-io/kepler-model-server/pull/206 (cc @sunya-ch @marceloamaral )

**Comment by jiere** [comment]
@rootfs , as we talked offline, I did not see similar issue in my local setup. 
It looks like this AWS instance is `Broadwell` platform, but the CPU model (Xeon E5-2686 v4) is not found in Intel `Broadwell` series [shipping SKUs](https://ark.intel.com/content/www/us/en/ark/products/codename/38530/products-formerly-broadwell.html) list,  please help confirm your conclusion and help arrange a live debug, thanks.

Below are my local try logs:
```
$ lscpu
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      52 bits physical, 57 bits virtual
CPU(s):                             112
On-line CPU(s) list:                0-111
Thread(s) per core:                 2
Core(s) per socket:                 28
Socket(s):                          2
NUMA node(s):                       2
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              106
Model name:                         Intel(R) Xeon(R) Gold 6348 CPU @ 2.60GHz
Stepping:                           6
CPU MHz:                            2600.000
CPU max MHz:                        3500.0000
CPU min MHz:                        800.0000
...
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
$ uname -r
5.15.0-89-generic
$ grep RAPL /boot/config-5.15.0-89-generic
CONFIG_PERF_EVENTS_INTEL_RAPL=m
CONFIG_PROC_THERMAL_MMIO_RAPL=m
CONFIG_INTEL_RAPL_CORE=m
CONFIG_INTEL_RAPL=m
$ ls /sys/class/powercap/intel-rapl
enabled  intel-rapl:0  intel-rapl:1  power  subsystem  uevent

```

**Reply by rootfs** [reply]
@jiere here is what i saw on the aws i3.metal instance

https://asciinema.org/a/626455

**Reply by rootfs** [reply]
Found a solution:

```sh
kernel_version=$(uname -r); apt install linux-modules-$kernel_version linux-modules-extra-$kernel_version
modprobe intel_rapl_common
```
Then we can use the powercap sysfs:
```console
root@ip-172-31-5-104:/home/ubuntu# ls /sys/class/powercap
intel-rapl  intel-rapl:0  intel-rapl:0:0  intel-rapl:1  intel-rapl:1:0
```

**Reply by jiere** [reply]
This should be a good solution.
